webapp/javascript/components/ServiceDiscoveryApp.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { fetchServiceDiscoveryAppData } from '../redux/actions';
+
+type Target = {
+  DiscoveredLabels: Record<string, string>;
+  Labels: Record<string, string>;
+  Job: string;
+  Url: string;
+  GlobalUrl: string;
+  LastError: string;
+  LastScrape: string;
+  LastScrapeDuration: number;
+  Health: string;
+};
+
+const ServiceDiscoveryApp = (props) => {
+  const { actions, data, url } = props;
+  const [groups, setGroups] = useState({});
+
+  useEffect(() => {
+    actions.fetchServiceDiscoveryAppData(url);
+  }, []);
+
+  useEffect(() => {
+    if (data) {
+      const acc = data.reduce((acc, next) => {
+        if (!acc[next.Job]) {
+          acc[next.Job] = [];
+        }
+        acc[next.Job].push(next);
+        return acc;
+      }, {});
+      setGroups(acc);
+    }
+  }, [data]);
+
+  function getUpCount(targets: Target[]) {
+    return targets.filter((t) => t.Health === 'up').length;
+  }
+  return (
+    <div
+      className="p-2"
+      style={{
+        fontSize: '12px',
+        width: '100%',
+        height: '100%',
+        overflow: 'auto',
+        margin: '0 0 0 20px',
+      }}
```

Comment:
we're trying to switch to this pattern for styles:
* you put all styles in a module.scss file https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/components/ExportData.module.scss
* then you import it from your component https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/components/ExportData.tsx#L11
* then you use classes by names https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/components/ExportData.tsx#L247

Reply:
Please ignore the inline styles. I've purposefully left it this way so I could iterate on the design faster!